ambassador/ambassador/ir/ircluster.py

```diff
@@ -169,7 +172,13 @@ def __init__(self, ir: 'IR', aconf: Config,
         # p is read-only, so break stuff out.
 
         hostname = p.hostname
-        port = p.port
+        try:
```

Comment:
This doesn't feel like the right spot -- surely it's the `urlparse` that fails, not the port assignment?

Reply:
Even I was surprised that it's the port assignment that fails. Posting the following from /usr/lib64/python3.7/urllib/parse.py -
```python
    @property
    def port(self):
        port = self._hostinfo[1]
        if port is not None:
            port = int(port, 10)
            if not ( 0 <= port <= 65535):
                raise ValueError("Port out of range 0-65535")
        return port
```

Reply:
...oooookay then. :)